src/main/cluster-manager.ts

```diff
@@ -161,14 +163,22 @@ export class ClusterManager extends Singleton {
       const cluster = this.store.getById(entity.metadata.uid);
 
       if (!cluster) {
-        this.store.addCluster({
-          id: entity.metadata.uid,
-          preferences: {
-            clusterName: entity.metadata.name
-          },
-          kubeConfigPath: entity.spec.kubeconfigPath,
-          contextName: entity.spec.kubeconfigContext
-        });
+        try {
+          this.store.addCluster({
+            id: entity.metadata.uid,
+            preferences: {
+              clusterName: entity.metadata.name
+            },
+            kubeConfigPath: entity.spec.kubeconfigPath,
+            contextName: entity.spec.kubeconfigContext
+          });
+        } catch (error) {
```

Comment:
Does this `entity` need to be removed from `entities`, or somehow signal that it should be removed?

Reply:
If the file is no longer present then kubesync will detect that it is no longer there are remove it from the catalog. If kubesync tries to read a file and gets an EACCESS then it clears all entities that it had previously detected.